lib/stateView/document/mongoDbRepository/SVDocumentMongoDbRepository.js

```diff
@@ -11,18 +11,23 @@ const InvalidQueryError = require('../errors/InvalidQueryError');
 
 class SVDocumentMongoDbRepository {
   /**
-   * @param {Db} mongoClient
+   * @param {Db} db
    * @param {convertWhereToMongoDbQuery} convertWhereToMongoDbQuery
    * @param {validateQuery} validateQuery
    * @param {string} documentType
    */
-  constructor(mongoClient, convertWhereToMongoDbQuery, validateQuery, documentType) {
-    this.mongoClient = mongoClient.collection(`documents_${documentType}`);
+  constructor(db, convertWhereToMongoDbQuery, validateQuery, documentType) {
+    this.db = db;
+    this.mongoClient = db.collection(`documents_${documentType}`);
     this.convertWhereToMongoDbQuery = convertWhereToMongoDbQuery;
     this.validateQuery = validateQuery;
     this.documentType = documentType;
   }
 
+  async createCollection(documentType) {
```

Comment:
Use private method this.getCollectionName() which responds with `documents_${documentType}`